api/rest/context.go

```diff
@@ -1,17 +1,50 @@
 package rest
```

Comment:
rest is a package that deals with HTTP request handling, therefore placing here codes that do not deal with requests processing it is wrong, previos location i think was right